### PR TITLE
fix for node.js versions > 13

### PR DESCRIPTION
**Pull request by den1zk**
PollCtx::UnwrapPointer was causing crashes in Node.js versions greater than 13.x.

### CODE REVIEW

**Comment by m-ohuchi** [comment]
@nickdesaulniers 
We confirmed this fix work as expected. Is it possible to examine and proceed to merge?
Tested on Debian 10.3 with Node.js v14.18.1 and v16.13.0.

File: src/poll_ctx.cc [hunk 1]
```diff
@@ -35,6 +35,10 @@ v8::Local<v8::Value> PollCtx::WrapPointer (void* ptr, size_t length) {
...
 
 PollCtx* PollCtx::UnwrapPointer (v8::Local<v8::Value> buffer) {
-  return reinterpret_cast<PollCtx*>(node::Buffer::HasInstance(buffer) ?
-    node::Buffer::Data(buffer.As<v8::Object>()) : NULL);
+   v8::Local<v8::ArrayBufferView> ui = buffer.As<v8::ArrayBufferView>();
+   v8::Local<v8::ArrayBuffer> abuf = ui->Buffer();
+   std::shared_ptr<v8::BackingStore> ab_c = abuf->GetBackingStore();
+   abuf->Detach();
+   return reinterpret_cast<PollCtx*>(node::Buffer::HasInstance(buffer) ?
+      static_cast<char*>(ab_c->Data()) : NULL  );
```
Comment (nickdesaulniers) [inline src/poll_ctx.cc:43]: Please remove the trailing whitespace after `NULL`.

**Comment by den1zk** [comment]
Hi, I did it.

Nick Desaulniers ***@***.***>, 10 Ara 2021 Cum, 22:35
tarihinde şunu yazdı:

**Comment by pkedy** [comment]
@den1zk looks like there is still a trailing space.

@nickdesaulniers I confirmed this fixes a POC I have. I'd love to see this merged and a new npm release cut.

**Comment by nickdesaulniers** [comment]
thanks for the fix, merged in https://github.com/nickdesaulniers/node-nanomsg/commit/e5b1da1ea00cee78049e54d4b573f42e27d28ac6. Closing out this PR.

Will need @reqshark 's help to `npm publish` the latest version.  @reqshark can you add me to https://www.npmjs.com/package/nanomsg somehow?  I think if you go to https://www.npmjs.com/package/nanomsg/access and send [me](https://www.npmjs.com/~wangstabill) an invite, I should be able to publish.

**Comment by jonathan-roy** [comment]
Hi and thanks for your work, any news about that release?

**Comment by nickdesaulniers** [comment]
Sorry for the delays. So I think what's happened is that my npmjs.org account is tied to my previous employer's email address I had but no longer have access to. I've contact staff at npmjs.org for support/access to https://www.npmjs.com/package/nanomsg. Once we have that blunder of mine sorted out, I should be able to `npm publish` the 4.2.0 release.

In the meantime, you might be able to fetch from github rather than npm from your package.json file, IIRC.

Sorry again for the delays!

**Comment by nickdesaulniers** [comment]
I've been in contact with npmjs support staff on recovering my npmjs account in order to publish the latest release. Still a WIP.

File: src/poll_ctx.cc (hunk 1 above)
Comment (den1zk) [reply src/poll_ctx.cc]: Hi, 
I removed the character.

**Comment by nickdesaulniers** [comment]
[published](https://www.npmjs.com/package/nanomsg/v/4.2.0)

**Pull request closed without merging**